### PR TITLE
Feat(parserApi)[OCT-27]: basic implementation of parser api 

### PR DESCRIPTION
**Pull request by kogonidze**

### CODE REVIEW

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                handler: async (IMediator mediator, CancellationToken cancellationToken) =>
+                {
+                    var models = await mediator.Send(
+                        request: new GetAllParserDynamicEntityModelsQuery(),
+                        cancellationToken: cancellationToken);
+
+                    return Results.Ok(models);
+                })
+                .Produces<Response<IEnumerable<ParserDynamicEntityModel>>>(StatusCodes.Status200OK)
+                .Produces(StatusCodes.Status204NoContent)
```
Comment (VladiTitov) [inline src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs:18]: The description is not true, this method returns only status code 200.

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                handler: async (string id, IMediator mediator, CancellationToken cancellationToken) =>
+                {
+                    var model = await mediator.Send(
+                        request: new GetParserDynamicEntityModelByIdQuery() { Id = id },
+                        cancellationToken: cancellationToken);
+
+                    return Results.Ok(model);
+                })
+                .Produces<Response<ParserDynamicEntityModel>>(StatusCodes.Status200OK)
+                .Produces(StatusCodes.Status404NotFound)
```
Comment (VladiTitov) [inline src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs:33]: See above.

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                    IMediator mediator,
+                    CancellationToken cancellationToken) =>
+                {
+                    return await mediator.Send(
+                        request: command,
+                        cancellationToken: cancellationToken) is Response<string> response
+                        ? Results.Created($"api/parserDynamicEntityModel/{response.Data}", response.Data)
+                        : Results.BadRequest();
+                })
+                .Produces(StatusCodes.Status201Created)
```
Comment (VladiTitov) [inline src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs:49]: Here it is possible to return status code 400.

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                    IMediator mediator,
+                    CancellationToken cancellationToken) =>
+                {
+                    return await mediator.Send(
+                        request: command,
+                        cancellationToken: cancellationToken) is Response<IDictionary<int, string>> response
+                        ? Results.Created($"api/parserDynamicEntityModels/{response.Data}", response.Data)
+                        : Results.BadRequest();
+                })
+                .Produces(StatusCodes.Status201Created)
```
Comment (VladiTitov) [inline src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs:65]: See above.

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                {
+                    var result = await mediator.Send(
+                        request: new DeleteParserDynamicEntityModelByIdCommand() { Id = id },
+                        cancellationToken: cancellationToken);
+
+                    return result.Data
+                        ? Results.NoContent()
+                        : Results.NotFound();
+                })
+                .Produces(StatusCodes.Status204NoContent)
```
Comment (VladiTitov) [inline src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs:100]: See above.

File: src/Services/Parser/Api/Core/Parser.Core.Application/Queries/GetParserDynamicEntityModelById/GetParserDynamicEntityModelByIdQueryValidator.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    public class GetParserDynamicEntityModelByIdQueryValidator 
+        : AbstractValidator<GetParserDynamicEntityModelByIdQuery>
+    {
+        public GetParserDynamicEntityModelByIdQueryValidator()
+        {
+            var objectIdLengthConstraint = 24;
+
+            RuleFor(query => query.Id)
+                .NotEmpty()
+                .Length(objectIdLengthConstraint);
```
Comment (VladiTitov) [inline src/Services/Parser/Api/Core/Parser.Core.Application/Queries/GetParserDynamicEntityModelById/GetParserDynamicEntityModelByIdQueryValidator.cs:12]: I see you are using the global ErrorHandlerMiddleware, maybe you should accompany the validation with more meaningful messages?

File: src/Services/Parser/Api/Core/Parser.Core.Application/Commands/ReplaceOneParserDynamicEntityModel/ReplaceOneParserDynamicEntityModelCommandValidator.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        public ReplaceOneParserDynamicEntityModelCommandValidator()
+        {
+            var objectIdLengthConstraint = 24;
+
+            RuleFor(command => command).NotEmpty();
+            RuleFor(command => command.Id)
+                .NotEmpty()
+                .Length(objectIdLengthConstraint);
+            RuleFor(command => command.EntityName).NotEmpty();
+            RuleFor(command => command.Properties).NotEmpty();
```
Comment (VladiTitov) [inline src/Services/Parser/Api/Core/Parser.Core.Application/Commands/ReplaceOneParserDynamicEntityModel/ReplaceOneParserDynamicEntityModelCommandValidator.cs:15]: This validation allows me to create an entity with empty nested fields, is that by design?

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs (hunk 1 above)
Comment (kogonidze) [reply src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs:18]: Sure, fixed

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs (hunk 4 above)
Comment (kogonidze) [reply src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs:65]: fixed

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs (hunk 5 above)
Comment (kogonidze) [reply src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs:100]: fixed

**Review by Anton-Sam — APPROVED** [review]
Good job!

File: src/Services/Parser/Api/Core/Parser.Core.Application/Commands/ReplaceOneParserDynamicEntityModel/ReplaceOneParserDynamicEntityModelCommandValidator.cs (hunk 7 above)
Comment (kogonidze) [reply src/Services/Parser/Api/Core/Parser.Core.Application/Commands/ReplaceOneParserDynamicEntityModel/ReplaceOneParserDynamicEntityModelCommandValidator.cs:15]: Of course it's unexpected behaviour. Fixed

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs (hunk 2 above)
Comment (kogonidze) [reply src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs:37]: fixed

File: src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs (hunk 3 above)
Comment (kogonidze) [reply src/Services/Parser/Api/Presentation/Parser.Presentation.WebApi/Endpoints/DynamicEntityEndpoints.cs]: fixed

**Review by VladiTitov — APPROVED** [review]
Good job! Well done, Vasily!

**Pull request merged**